js/ad_manager_vast.js

```diff
@@ -801,7 +786,32 @@ OO.Ads.manager(function(_, $) {
      * @method Vast#buildTimeline
      */
     this.buildTimeline = function() {
-      return null;
+      var timeline = [];
+      if (this.allAdInfo && _.isArray(this.allAdInfo))
+      {
+        for (var i = 0; i < this.allAdInfo.length; i++)
+        {
+          var ad = this.allAdInfo[i];
+          //use linear overlay as fake ad so we don't have to specify stream type.
+          var adData = {
+            adManager: this.name,
+            ad: ad,
+            duration: 0,
+            adType: this.amc.ADTYPE.LINEAR_OVERLAY,
+          };
+
+          if (ad.position_type == 't') {
+            adData.position = ad.time/1000;
+          } else if (ad.position_type == 'p') {
+            //TODO
+            //adData.position =
+          }
+          var amcAd = new this.amc.Ad(adData);
+          amcAd.isAdRequest = true;
```

Comment:
The IMA ad manager uses an ad request type "adRequest" to mark an ad as an ad request. We can do the same thing here to maintain consistency across ad managers.